src/mono/wasm/runtime/marshal-to-cs.ts

```diff
@@ -49,103 +50,38 @@ export function initialize_marshalers_to_cs(): void {
-    let converter_name_arg3 = "null";
-    let converter_name_res = "null";
-
-    let marshaler_type = get_signature_type(sig);
+export function bind_arg_marshal_to_cs(sig: JSMarshalerType, marshaler_type: MarshalerType, index: number): BoundMarshalerToCs | undefined {
     if (marshaler_type === MarshalerType.None || marshaler_type === MarshalerType.Void) {
-        return {
-            converters,
-            call_body,
-            marshaler_type
-        };
+        return undefined;
     }
+    let res_marshaler: MarshalerToCs | undefined = undefined;
+    let arg1_marshaler: MarshalerToJs | undefined = undefined;
+    let arg2_marshaler: MarshalerToJs | undefined = undefined;
+    let arg3_marshaler: MarshalerToJs | undefined = undefined;
 
+    arg1_marshaler = get_marshaler_to_js_by_type(get_signature_arg1_type(sig));
+    arg2_marshaler = get_marshaler_to_js_by_type(get_signature_arg2_type(sig));
+    arg3_marshaler = get_marshaler_to_js_by_type(get_signature_arg3_type(sig));
     const marshaler_type_res = get_signature_res_type(sig);
-    if (marshaler_type_res !== MarshalerType.None) {
-        const converter = js_to_cs_marshalers.get(marshaler_type_res);
-        mono_assert(converter && typeof converter === "function", () => `Unknow converter for type ${marshaler_type_res} at ${index}`);
-
-
-        if (marshaler_type != MarshalerType.Nullable) {
-            converter_name_res = "converter" + index + "_res";
-            converters += ", " + converter_name_res;
-            converter_types += " " + MarshalerType[marshaler_type_res];
-            closure[converter_name_res] = converter;
-        }
-        else {
-            marshaler_type = marshaler_type_res;
-        }
-    }
-
-    const marshaler_type_arg1 = get_signature_arg1_type(sig);
-    if (marshaler_type_arg1 !== MarshalerType.None) {
-        const converter = cs_to_js_marshalers.get(marshaler_type_arg1);
-        mono_assert(converter && typeof converter === "function", () => `Unknow converter for type ${marshaler_type_arg1} at ${index}`);
-
-        converter_name_arg1 = "converter" + index + "_arg1";
-        converters += ", " + converter_name_arg1;
-        converter_types += " " + MarshalerType[marshaler_type_arg1];
-        closure[converter_name_arg1] = converter;
-    }
-
-    const marshaler_type_arg2 = get_signature_arg2_type(sig);
-    if (marshaler_type_arg2 !== MarshalerType.None) {
-        const converter = cs_to_js_marshalers.get(marshaler_type_arg2);
-        mono_assert(converter && typeof converter === "function", () => `Unknow converter for type ${marshaler_type_arg2} at ${index}`);
-
-        converter_name_arg2 = "converter" + index + "_arg2";
-        converters += ", " + converter_name_arg2;
-        converter_types += " " + MarshalerType[marshaler_type_arg2];
-        closure[converter_name_arg2] = converter;
+    res_marshaler = get_marshaler_to_cs_by_type(marshaler_type_res);
+    if (marshaler_type === MarshalerType.Nullable) {
```

Comment:
can you add a comment explaining what this is about? It's not clear to me

Reply:
Added `// nullable has nested type information, it's stored in res slot of the signature. The marshaler is the same as for non-nullable primitive type.`